### PR TITLE
HMAN-89 - Update /hearing (Get) - response Payload

### PR DESCRIPTION
**Pull request by markdathornehmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-89

### Change description ###

Update Get hearing response Payload as per given detail.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by RebeccaBaker — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import uk.gov.hmcts.reform.hmc.data.LinkedHearingDetailsAudit;
+
+import java.util.List;
+
+@Transactional(propagation = Propagation.REQUIRES_NEW)
+@Repository
+public interface LinkedHearingDetailsRepository extends CrudRepository<LinkedHearingDetailsAudit, Long> {
+
+    @Query("from LinkedHearingDetailsAudit lhd WHERE lhd.hearing.id = :hearingId ")
+    LinkedHearingDetailsAudit getLinkedHearingDetailsByHearingId(Long hearingId);
```
Comment (padmapriyanalam) [inline src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java:17]: The requirement says  to get the details from **linkedHearingDetails** table. the query here use LinkedHearingDetailsAudit table

File: src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Query("from LinkedHearingDetailsAudit lhd WHERE lhd.linkedGroup.requestId = :requestId ")
+    List<LinkedHearingDetailsAudit> getLinkedHearingDetailsByRequestId(String requestId);
```
Comment (padmapriyanalam) [inline src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java:20]: No usages found for this query in the source code

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapper.java [hunk 3]
```diff
@@ -167,52 +191,60 @@ private OrganisationDetails setOrganisationDetails(HearingPartyEntity hearingPar
...
             hearingResponse.setListAssistTransactionID(
-                hearingResponseEntity.getHearingResponseId());
+                 hearingResponseEntity.getListAssistTransactionId());
             hearingResponse.setReceivedDateTime(hearingResponseEntity.getRequestTimeStamp());
             hearingResponse.setResponseVersion(hearingResponseEntity.getHearingResponseId());
-            hearingResponse.setLaCaseStatus(hearingResponseEntity.getListingCaseStatus());
-            hearingResponse.setListingStatus(hearingResponseEntity.getListingStatus());
+            hearingResponse.setLaCaseStatus(ListAssistCaseStatus.getLabel(
+                    hearingResponseEntity.getListingCaseStatus()));
+            hearingResponse.setListingStatus(ListingStatus.getLabel(hearingResponseEntity.getListingStatus()));
```
Comment (padmapriyanalam) [inline src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapper.java:247]: hearingCancellationReason value is not populated. Should be retrieved from hearingResponse.cancellationReasonType

File: src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java (hunk 1 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java:17]: The linkedHearingDetails table is obsolete. There is only the audit table now.

File: src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java (hunk 2 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/repository/LinkedHearingDetailsRepository.java]: Removed.

File: src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapper.java (hunk 3 above)
Comment (markdathornehmcts) [reply src/main/java/uk/gov/hmcts/reform/hmc/helper/GetHearingResponseMapper.java:247]: Code updated. Thanks.

**Review by padmapriyanalam — APPROVED** [review]

**Pull request closed without merging**